### PR TITLE
Added health monitor gui, network vitals, app to view monitor

### PR DESCRIPTION
**Pull request by AgoinsUSGS**
Created NetworkVitals object that contains Control Network health information.
Created ControlHealthMonitor class to display all information from the NetworkVitals object.
Created temporary application "healthmonitor" to display these two classes.

### CODE REVIEW

**Comment by AgoinsUSGS** [comment]
Hold off on merging this for now. Gotta push something up before merge

**Comment by AgoinsUSGS** [comment]
This can be reviewed and merged now.

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [inline isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp]: This will need to return a QList of QLists of QStrings, the internal lists are the islands.

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [inline isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp]: You should use GetType and then compare against ControlPoint::Fixed instead of getting it as a string.

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [inline isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp]: You should use GetType and then compare against ControlPoint::Constrained instead of getting it as a string.

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [inline isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp:84]: You should use GetType and then compare against ControlPoint::Free instead of getting it as a string.

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [inline isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp:126]: Is there a reason why you would use this instead of the ControlNet's method?

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [inline isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp:130]: Same as getCubeSerials, why not just call the ControlNet's method. Do you see a usage where the caller doesn't have access to the network?

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [inline isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp]: Same as the counting methods, use the actual enumeration instead of string comparisons.

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [inline isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp]: Don't use 1 line if statements. They are clean, but they obscure the fact that there's a check inside of the foreach loop.

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [inline isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp:221]: This method looks like GUI functionality. Is there a reason why it's not in the GUI?

**Review by jessemapel — CHANGES_REQUESTED** [review]

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (AgoinsUSGS) [reply isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp:126]: The GUI doesn't interface with a ControlNet directly to grab the data it's representing. It interfaces with a ControlNetVital's object only, so we use a wrapper around the network in ControlNetVitals to grab data like this.

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (AgoinsUSGS) [reply isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp:221]: I felt like the overall status of a network was relevant information to store in the NetworkVitals object to reflect the current sitting status of it's Control Network. This slot is connected to the netModified() slot of the ControlNet to recalculate the overall status of the network. If we decide that the status and health of a network should be determined by the GUI then I can absolutely move it there

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [reply isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp:126]: :+1:

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [reply isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp:221]: Sounds good

File: isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp
Comment (jessemapel) [inline isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp]: This looks like a typo, probably should compare against ControlPoint::Free

**Review by jessemapel — CHANGES_REQUESTED** [review]
1 minor typo to fix still

**Comment by AgoinsUSGS** [comment]
Just noticed, I had been using "Constrainted" instead of "Constrained" as well. fixed that too.

**Review by jessemapel — APPROVED** [review]

**Pull request merged**
